src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/repositories/StudyRepository.java

```diff
@@ -17,24 +17,24 @@
  */
 package uk.ac.ebi.ampt2d.metadata.persistence.repositories;
 
+import org.springframework.data.querydsl.QueryDslPredicateExecutor;
+import org.springframework.data.querydsl.binding.QuerydslBinderCustomizer;
+import org.springframework.data.querydsl.binding.QuerydslBindings;
 import org.springframework.data.repository.CrudRepository;
-import org.springframework.data.repository.query.Param;
 import org.springframework.data.rest.core.annotation.RepositoryRestResource;
-import org.springframework.data.rest.core.annotation.RestResource;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.QStudy;
 import uk.ac.ebi.ampt2d.metadata.persistence.entities.Study;
 
-import java.util.List;
-
 @RepositoryRestResource
-public interface StudyRepository extends CrudRepository<Study, Long> {
-
-    @RestResource(path = "findByAssemblyName")
-    List<Study> findByAnalyses_Assembly_Name(@Param("name") String name );
+public interface StudyRepository extends CrudRepository<Study, Long>,
+        QueryDslPredicateExecutor<Study>, QuerydslBinderCustomizer<QStudy> {
 
-    @RestResource(path = "findByAssemblyNameAndPatch")
-    List<Study> findByAnalyses_Assembly_NameAndAnalyses_Assembly_Patch(
-            @Param("name") String name,
-            @Param("patch") String patch
-    );
+    default void customize(QuerydslBindings bindings, QStudy study) {
+        bindings.bind(study.analyses.any().assembly.name,
+                study.analyses.any().assembly.patch)
+                .first((path, value) -> path.equalsIgnoreCase(value));
+        bindings.bind(study.analyses.any().type)
+                .first((path, value) -> path.eq(value));
```

Comment:
By default it checks for equality , so we need to bind only if we need extra conditions like equalsIgnorecase, i think.

Reply:
Good point! Thanks!